### PR TITLE
Infinity mode

### PR DESCRIPTION
**Pull request by bink-git**

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -1,2 +1,3 @@
 export const API_URL =
-  'https://opentdb.com/api.php?amount=10&category=9&difficulty=easy&type=multiple';
+  'https://opentdb.com/api.php?amount=50&category=9&difficulty=easy&type=multiple';
+export const TOKEN = 'https://opentdb.com/api_token.php?command=request';
```
Comment (AgileNix) [inline src/constants.js]: Rename this constant.

It's not a token, but it's a URL that you can use to get a token.

File: src/constants.js [hunk 2]
```diff
@@ -1,2 +1,3 @@
 export const API_URL =
-  'https://opentdb.com/api.php?amount=10&category=9&difficulty=easy&type=multiple';
+  'https://opentdb.com/api.php?amount=50&category=9&difficulty=easy&type=multiple';
```
Comment (AgileNix) [inline src/constants.js]: Extract common parts of the URLs like `https://opentdb.com/api.php` to avoid code duplication

File: src/App.jsx [hunk 3]
```diff
@@ -33,29 +37,41 @@ function App() {
...
-        const res = await axios.get(API_URL);
+        const userToken = await axios.get(TOKEN);
+        const { token, response_code } = userToken.data;
+        setToken(token);
+        setResCode(response_code);
+        const res = await axios.get(`${API_URL}&token=${token}`);
         setData(res.data.results);
       } catch (error) {
-        notify();
+        if (resCode === 3) {
```
Comment (AgileNix) [inline src/App.jsx]: It's unclear what's the purpose of the number `3` here.

If it's something given to you by the API, then make it a named constant.

File: src/App.jsx [hunk 4]
```diff
@@ -33,29 +37,41 @@ function App() {
...
   };
 
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const res = await axios.get(API_URL);
+        const userToken = await axios.get(TOKEN);
+        const { token, response_code } = userToken.data;
+        setToken(token);
+        setResCode(response_code);
```
Comment (AgileNix) [inline src/App.jsx]: Can you find a way to simplify the code by removing `setResCode`?

File: src/App.jsx [hunk 5]
```diff
@@ -33,29 +37,41 @@ function App() {
...
     setCorrectAnswers(0);
     setStart(true);
+    setShowResult(false);
   };
 
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const res = await axios.get(API_URL);
+        const userToken = await axios.get(TOKEN);
```
Comment (AgileNix) [inline src/App.jsx]: The user token should be received only once when the user starts playing. All future requests should reuse the same token until it expires.

**Review by AgileNix — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 5 above)
Comment (bink-git) [reply src/App.jsx]: I'm not sure that I wrote it correctly

File: src/App.jsx (hunk 4 above)
Comment (bink-git) [reply src/App.jsx]: I tried via variables but not sure

File: src/App.jsx [hunk 6]
```diff
@@ -17,8 +17,21 @@ function App() {
...
+  const [token, setToken] = useState('');
+  // const [resCode, setResCode] = useState('');
 
-  const notify = () => toast.error('Error fetching data');
+  const { code, message } = RESPONSE_CODES;
+
+  let resCode;
+  let resMessage;
+
+  // const { code, message } = RESPONSE_CODES.find(
```
Comment (AgileNix) [inline src/App.jsx]: remove commented code

if it's something valuable you're afraid to lose, then push it into a separate branch or stash it locally

File: src/App.jsx [hunk 7]
```diff
@@ -17,8 +17,21 @@ function App() {
   const [correctAnswers, setCorrectAnswers] = useState(0);
   const [start, setStart] = useState(true);
   const [reset, setReset] = useState(true);
+  const [showResult, setShowResult] = useState(false);
+  const [token, setToken] = useState('');
+  // const [resCode, setResCode] = useState('');
```
Comment (AgileNix) [inline src/App.jsx]: there's no place for commented code in the finished PRs

**Review by AgileNix — CHANGES_REQUESTED** [review]

**Pull request merged**